benches/payout-stakers/src/main.rs

```diff
@@ -200,37 +239,69 @@ fn nominate_validator(
         .for_each(|chunk| staking_batch_nominate(connection, chunk));
 }
 
-fn bond_validate(address: &str, validators: Vec<KeyPair>) -> Vec<KeyPair> {
-    staking_multi_bond(address, &validators, MIN_VALIDATOR_BOND);
-    validators.par_iter().for_each(|account| {
+/// Bonds controller accounts to the corresponding validator accounts.
+/// We assume stash == validator != controller.
+fn bond_controllers_and_validators(
+    address: &str,
+    controllers: Vec<KeyPair>,
+    validators: Vec<KeyPair>,
+) {
+    controllers
+        .into_par_iter()
+        .zip(validators.into_par_iter())
+        .for_each(|(controller, validator)| {
+            let connection = SignedConnection::new(address, validator);
+            let controller_account_id = AccountId::from(controller.public());
+            staking_bond(
+                &connection,
+                MIN_VALIDATOR_BOND,
+                &controller_account_id,
+                XtStatus::InBlock,
+            );
+        });
+}
+
+/// Validate the controller and validator pairs.
+/// We assume stash == validator != controller.
+fn validate_controllers_and_validators(address: &str, controllers: Vec<KeyPair>) {
+    controllers.par_iter().for_each(|controller| {
         let mut rng = thread_rng();
-        let connection = SignedConnection::new(address, account.clone());
+        let connection = SignedConnection::new(address, controller.clone());
         staking_validate(&connection, rng.gen::<u8>() % 100, XtStatus::InBlock);
     });
-    validators
 }
 
+/// For a specific validator given by index, generates a predetermined number of nominator accounts.
+/// Nominator accounts are produced as (controller, stash) tuples with initial endowments.
 fn generate_nominator_accounts_with_minimal_bond(
     connection: &SignedConnection,
     validator_number: u32,
     validators_count: u32,
-) -> Vec<AccountId> {
+) -> (Vec<AccountId>, Vec<AccountId>) {
     info!(
         "Generating nominator accounts for validator {}",
         validator_number
     );
-    let accounts = (0..NOMINATOR_COUNT)
-        .map(|nominator_number| {
-            derive_user_account_from_numeric_seed(
-                validators_count + nominator_number + NOMINATOR_COUNT * validator_number,
-            )
-        })
-        .map(|key_pair| AccountId::from(key_pair.public()))
-        .collect::<Vec<_>>();
-    accounts
+    let mut controller_accounts = vec![];
+    let mut stash_accounts = vec![];
+    (0..NOMINATOR_COUNT).for_each(|nominator_number| {
+        let idx = validators_count + nominator_number + NOMINATOR_COUNT * validator_number;
+        let controller = keypair_from_string(&format!("//{}//Controller", idx));
+        let stash = keypair_from_string(&format!("//{}//Stash", idx));
```

Comment:
OK, on to the big thing. I am aware that this is less then ideal. It is the actual root cause behind this PR. AFAIK, until recently we did not have a unified procedure for generating accounts for various kinds of runs. @Marcin-Radecki has made very good progress on fixing that. At the moment, we still rely on the seeds being consistent across different parts of the code (e.g. chain spec and benches). I would rather keep the scope of this PR limited and potentially raise another one to address this as it would require some consideration.

Reply:
I was not aware there are some external contraints that force it to look like that. But in future please mind that `//` derivation is designed for human convenience and handles any string. Doing things like `//0`, `//1` is a horrible idea and explicit asking for conflicts.